lib/fibonacci.rb

```diff
@@ -1,9 +1,18 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n) - number of cycles correlates directly with size of n
+# Space Complexity - O(n) - array is proportional to n
 # Hint, you may want a recursive helper method
 # Hint:  Fib(0) = 0, Fib(1) = 1, work up from there
 def fibonacci(n)
```

Comment:
This works, but can you see a way to do it without needing an entire array from 0..n in size?